### PR TITLE
SYSE-388/Fix-API-and-UI-tests

### PR DESCRIPTION
**Pull request by ermirizio**

### CODE REVIEW

**Review by alephnull — APPROVED** [review]

File: policy/templates/subtemplates/auto/auto-test.gotmpl [hunk 1]
```diff
@@ -149,9 +151,9 @@
           python-version: '3.10'
       - name: Run API tests
         id: test_execution
-        working-directory: tests/{{ .test }}
+        working-directory: tyk-analytics/tests/{{ .test }}
         run: |
-          pytest="pytest --random-order --maxfail=3 --junitxml=${XUNIT_REPORT_PATH} --cache-clear --log-cli-level=ERROR"
+          pytest="pytest --ci --random-order --force-flaky --no-success-flaky-report --maxfail=3 --junitxml=${XUNIT_REPORT_PATH} --cache-clear --ignore=./tests/mdcb -v --log-cli-level=ERROR"
```
Comment (alephnull) [inline policy/templates/subtemplates/auto/auto-test.gotmpl:156]: Should we be ignoring mdcb tests?

File: policy/templates/subtemplates/auto/auto-test.gotmpl [hunk 2]
```diff
@@ -198,6 +202,7 @@
 
       - name: Generate metadata and upload test reports
         id: metadata_report
+        if: always() && (steps.test_execution.conclusion != 'skipped')
```
Comment (alephnull) [inline policy/templates/subtemplates/auto/auto-test.gotmpl:205]: Isn't this the same as always()? I think it should be just `steps.test_execution.conclusion != 'skipped'`

File: policy/templates/subtemplates/auto/auto-test.gotmpl (hunk 1 above)
Comment (ermirizio) [reply policy/templates/subtemplates/auto/auto-test.gotmpl:156]: this does not ignore mdcb tests, mdcb tests are located into mdcb_tests folder. /test/mdcb is used to import some base clases and old tests

File: policy/templates/subtemplates/auto/auto-test.gotmpl (hunk 2 above)
Comment (ermirizio) [reply policy/templates/subtemplates/auto/auto-test.gotmpl:205]: this combination will make it run only when test execution has been executed, wether failed or not by using always, `steps.test_execution.conclusion != 'skipped'` will prevent metadata report from running if tests were not executed due to a previous failure like env up

**Pull request merged**